components/org.wso2.identity.event.websubhub.publisher/src/main/java/org/wso2/identity/event/websubhub/publisher/internal/ClientManager.java

```diff
@@ -0,0 +1,145 @@
+        } catch (IOException e) {
+            throw WebSubHubAdapterUtil.handleServerException
+                    (WebSubHubAdapterConstants.ErrorMessages.ERROR_CREATING_ASYNC_HTTP_CLIENT, e);
+        }
+
+        RequestConfig config = createRequestConfig();
+        HttpAsyncClientBuilder httpClientBuilder = HttpAsyncClients.custom().setDefaultRequestConfig(config);
+        addSslContext(httpClientBuilder);
+        httpClientBuilder.setConnectionManager(connectionManager);
+        httpAsyncClient = httpClientBuilder.build();
+        httpAsyncClient.start();
+        LOG.info("HttpAsyncClient started");
+    }
+
+    /**
+     * Get HTTP client properly configured with tenant configurations.
+     *
+     * @return CloseableHttpAsyncClient instance.
+     */
+    public CloseableHttpAsyncClient getClient() throws WebSubAdapterException {
+
+        if (isNull(httpAsyncClient)) {
+            LOG.error("HttpAsyncClient is null");
+            throw WebSubHubAdapterUtil.handleServerException
+                    (WebSubHubAdapterConstants.ErrorMessages.ERROR_GETTING_ASYNC_CLIENT, null);
+        } else if (!httpAsyncClient.isRunning()) {
+            LOG.warn("HttpAsyncClient is not running, starting client");
+            httpAsyncClient.start();
+        }
+        return httpAsyncClient;
+    }
+
+    private RequestConfig createRequestConfig() {
+
+        return RequestConfig.custom()
+                .setConnectTimeout(WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration()
+                        .getHTTPConnectionTimeout())
+                .setConnectionRequestTimeout(WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration()
+                        .getHttpConnectionRequestTimeout())
+                .setSocketTimeout(
+                        WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration().getHttpReadTimeout())
+                .setRedirectsEnabled(false)
+                .setRelativeRedirectsAllowed(false)
+                .build();
+    }
+
+    private PoolingNHttpClientConnectionManager createPoolingConnectionManager() throws IOException {
+
+        int maxConnections = WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration()
+                .getDefaultMaxConnections();
+        int maxConnectionsPerRoute = WebSubHubAdapterDataHolder.getInstance().getAdapterConfiguration()
+                .getDefaultMaxConnectionsPerRoute();
+
+        ConnectingIOReactor ioReactor = new DefaultConnectingIOReactor();
+        PoolingNHttpClientConnectionManager poolingHttpClientConnectionMgr = new
+                PoolingNHttpClientConnectionManager(ioReactor);
+        // Increase max total connection to 20.
+        poolingHttpClientConnectionMgr.setMaxTotal(maxConnections);
+        // Increase default max connection per route to 20.
+        poolingHttpClientConnectionMgr.setDefaultMaxPerRoute(maxConnectionsPerRoute);
```

Comment:
Do we really want 20 to be opened for one endpoint ? IMO, for one end point we can keep the default of the http client which is 2

Reply:
Addressed with https://github.com/wso2-extensions/identity-event-publishers/pull/1/commits/f0ff52527687246da83cb9dbcb13ce640f0d6db3

Reply:
This comment stil says 20

Reply:
And IMO we don't need these two comments

Reply:
Addressed with https://github.com/wso2-extensions/identity-event-publishers/pull/1/commits/ec4d75aa968d6cbb73dd855225ccbadf7545d9fb